lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_retrospectives.py

```diff
@@ -239,6 +259,114 @@ def find_sample_origin_by_barcode(db: DatabaseSession, barcode: str) -> Optional
     return sample.sample_origin
 
 
+def create_encounter_hospitalization(redcap_record: dict) -> Optional[Dict[str, Dict]]:
+    """
+    Returns an Encounter.hospitalization entry created from a given *redcap_record*.
+    (https://www.hl7.org/fhir/encounter-definitions.html#Encounter.hospitalization)
+    """
+    disposition = discharge_disposition(redcap_record)
+
+    # For now, dischargeDisposition is the only info we store in
+    # Encounter.hospitalization. If this info isn't available, skip creating
+    # this resource entry.
+    if not disposition:
+        return None
+
+    return {
+        "dischargeDisposition": create_codeable_concept(
+            system = 'http://hl7.org/fhir/ValueSet/encounter-discharge-disposition',
+            code = disposition,
+        )
+    }
+
+
+def discharge_disposition(redcap_record: dict) -> Optional[str]:
+    """
+    Given a *redcap_record*, returns the mapped FHIR
+    Encounter.hospitalization.dischargeDisposition code
+    (https://www.hl7.org/fhir/valueset-encounter-discharge-disposition.html)
+    """
+    disposition = redcap_record['discharge_disposition']
+    if not disposition:
+        return None
+
+    if disposition.startswith('Disch/Trans/Planned IP Readm'):
+        # This feels like sensitive information. Don't code the entire string.
+        return 'other-hcf'
+
+    mapper = {
+        'HOME/SELF CARE': 'home',
+        'HOME HEALTH CARE': 'home',
+        'Transfer to Hospital': 'other-hcf',
+        'ICF- INTERMEDIATE CARE FACILITY': 'other-hcf',
+        'AGAINST MEDICAL ADVICE': 'aadvice',
+        'Expired': 'exp',
+        'Disch/Trans to a distinct Psych Unit/Hospital': 'psy',
+        'Disch/Trans to a distinct Rehab Unit/Hospital': 'rehab',
+        'SNF-SKILLED NURSING FACILITY': 'snf',
+        'DISCH/TRANS TO COURT/LAW ENFORCEMENT': 'oth',
+        'Other Institution - Not Defined Elsewhere': 'oth',
+    }
+
+    if disposition not in mapper:
+        raise Exception(f"Unknown discharge disposition value «{disposition}».")
+
+    return mapper[disposition]
```

Comment:
Easy enough to fix. I stripped leading/trailing white space and cast to lowercase before mapping. I'm not sure it makes sense to remove all spaces, though, because our codes are all either one word/initialism or are longer strings of text.

Reply:
Ah, not removing all spaces, but normalizing it for comparison, e.g. from the example I linked to:

```python
    def standardize_whitespace(string):
        return re.sub(r"\s+", " ", string.strip())
```

Reply:
Ah I misread the function. I can standardize white space, too.